### PR TITLE
Added a mapflag to block BANK

### PR DESCRIPTION
**Pull request by Pokye**
* **Server Mode**: Both

* **Description of Pull Request**: Thanks for @Lemongrass3110 I based it on your commit. ( https://github.com/rathena/rathena/pull/6291 )

### CODE REVIEW

File: conf/msg_conf/map_msg.conf [hunk 1]
```diff
@@ -930,7 +930,10 @@
 // @reloadbarterdb
 830: Barter database has been reloaded.
 
-//831-899 free
+// NoBank Mapflag
+831: You cannot use Bank on this map.
```
Comment (aleos89) [inline conf/msg_conf/map_msg.conf:934]:
```suggestion
831: You cannot use the Bank on this map.
```

File: src/map/clif.cpp [hunk 2]
```diff
@@ -7487,6 +7487,10 @@ void clif_parse_BankOpen(int fd, map_session_data* sd) {
 		clif_messagecolor(&sd->bl,color_table[COLOR_RED],msg_txt(sd,1496),false,SELF); //Banking is disabled
 		return;
 	}
+	if( map_getmapflag( sd->bl.m, MF_NOBANK ) ){
+		clif_displaymessage( sd->fd, msg_txt( sd, 831 ) ); // You cannot use Bank on this map.
```
Comment (aleos89) [inline src/map/clif.cpp:7491]:
```suggestion
		clif_displaymessage( sd->fd, msg_txt( sd, 831 ) ); // You cannot use the Bank on this map.
```

File: src/map/clif.cpp [hunk 3]
```diff
@@ -7558,6 +7562,10 @@ void clif_parse_BankCheck(int fd, map_session_data* sd) {
...
```
Comment (aleos89) [inline src/map/clif.cpp:7566]:
```suggestion
		clif_displaymessage( sd->fd, msg_txt( sd, 831 ) ); // You cannot use the Bank on this map.
```

File: src/map/clif.cpp [hunk 4]
```diff
@@ -7594,6 +7602,10 @@ void clif_parse_BankDeposit(int fd, map_session_data* sd) {
...
```
Comment (aleos89) [inline src/map/clif.cpp:7606]:
```suggestion
		clif_displaymessage( sd->fd, msg_txt( sd, 831 ) ); // You cannot use the Bank on this map.
```

File: src/map/clif.cpp [hunk 5]
```diff
@@ -7633,6 +7645,10 @@ void clif_parse_BankWithdraw(int fd, map_session_data* sd) {
...
```
Comment (aleos89) [inline src/map/clif.cpp:7649]:
```suggestion
		clif_displaymessage( sd->fd, msg_txt( sd, 831 ) ); // You cannot use the Bank on this map.
```

**Review by aleos89 — APPROVED** [review]

**Comment by CairoLee** [comment]
- Consider adding a flag info to the `mapinfo` atcommand
- Please add the relevant notes in `doc/mapflags.txt`

**Pull request merged**